Formula/m/magic-wormhole.rs.rb

```diff
@@ -0,0 +1,18 @@
+class MagicWormholeRs < Formula
+  desc "Rust implementation of Magic Wormhole, with new features and enhancements"
+  homepage "https://github.com/magic-wormhole/magic-wormhole.rs"
+  url "https://github.com/magic-wormhole/magic-wormhole.rs/archive/refs/tags/0.6.1.tar.gz"
+  sha256 "522db57161bb7df10feb4b0ca8dec912186f24a0974647dc4fccdd8f70649f96"
+  license "EUPL-1.2"
+  head "https://github.com/magic-wormhole/magic-wormhole.rs.git", branch: "master"
+
+  depends_on "rust" => :build
+
+  def install
+    system "cargo", "install", *std_cargo_args(path: "cli")
+  end
+
+  test do
+    assert_match version.to_s, shell_output("#{bin}/wormhole-rs -V").each_line.first.strip
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Can't we just reuse the same test from the Python `magic-wormhole` formula?
https://github.com/Homebrew/homebrew-core/blob/b5e0499542b82115d160e4a516b1832e66c1dbd6/Formula/m/magic-wormhole.rb#L167-L180

Reply:
not directly. `wormhole-rs` has no `--text` flag. i will try to implement something similar.

Reply:
implemented a real test ;)